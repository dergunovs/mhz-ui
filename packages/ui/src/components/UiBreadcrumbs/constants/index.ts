export const LINKS = [
  { url: '/', title: 'Main' },
  { url: '/category', title: 'Categories' },
  { url: '/cpu', title: 'CPU' },
];

export const DEFAULT_COLOR = 'default';
export const DEFAULT_ARIA_LABEL = 'Breadcrumb';
