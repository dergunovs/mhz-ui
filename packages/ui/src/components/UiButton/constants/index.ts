export const DEFAULT_LAYOUT = 'primary';
export const DEFAULT_TYPE = 'button';
export const DEFAULT_ICON = undefined;

export const DEFAULT_SLOT = 'Button label';
export const DEFAULT_ARIA_LABEL = 'Button';
