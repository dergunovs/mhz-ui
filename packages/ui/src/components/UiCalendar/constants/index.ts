import { ICalendarEvent } from '../interface';

export const EVENTS: ICalendarEvent<object>[] = [
  {
    id: 'id of event',
    start: new Date(),
    end: new Date(),
    title: '+',
    content: [{ id: 1, text: 'Text' }],
    color: 'linear-gradient(135deg, rgb(168, 60, 255) 36%, rgb(235, 0, 123) 69%, rgb(251, 185, 0) 100%)',
  },
  {
    id: 'id of event',
    start: new Date(),
    end: new Date(),
    title: '+',
    content: [{ id: 2, text: 'Text 2' }],
    color: 'linear-gradient(135deg, rgb(168, 60, 255) 41%, rgb(235, 0, 123) 71%, rgb(0, 134, 255) 100%)',
  },
];
