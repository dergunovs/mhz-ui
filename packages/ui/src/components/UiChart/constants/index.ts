export const LABELS = [
  '07.07 - 13.07',
  '14.07 - 20.07',
  '21.07 - 27.07',
  '28.07 - 03.08',
  '04.08 - 10.08',
  '11.08 - 17.08',
  '18.08 - 24.08',
];

export const DATASETS = [
  {
    data: [28, 16, 12, 4, 22, 16, 14],
    label: 'Грудь',
    borderColor: '#00CED1',
    backgroundColor: '#00CED1',
  },
  {
    data: [4, 6, 0, 2, 2, 0, 6],
    label: 'Ноги',
    borderColor: '#000080',
    backgroundColor: '#000080',
  },
  {
    data: [22, 18, 8, 10, 25, 18, 16],
    label: 'Плечи',
    borderColor: '#C41E3A',
    backgroundColor: '#C41E3A',
  },
  {
    data: [0, 6, 0, 6, 5, 4, 2],
    label: 'Пресс',
    borderColor: '#32CD32',
    backgroundColor: '#32CD32',
  },
  {
    data: [32, 24, 12, 10, 33, 24, 20],
    label: 'Руки',
    borderColor: '#DA70D6',
    backgroundColor: '#DA70D6',
  },
  {
    data: [16, 18, 8, 12, 26, 22, 20],
    label: 'Спина',
    borderColor: '#FF8C00',
    backgroundColor: '#FF8C00',
  },
];

export const TYPE = 'Line';
export const TITLE = 'Chart title';
