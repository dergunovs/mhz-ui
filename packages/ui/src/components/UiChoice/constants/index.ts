export const options = [1, 2, 3, 4, 5];
export const modelValue = options[2];
export const title = 'Заголовок';
export const isTall = true;
export const isInput = true;
