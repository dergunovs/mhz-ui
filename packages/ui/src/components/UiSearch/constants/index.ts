export const DEBOUNCE_TIME = 300;

export const MODEL_VALUE = '';
export const LANG = 'ru';
export const IS_SUCCESS = true;

export const RESULTS = {
  products: [
    { _id: '1', title: 'AMD Ryzen 7700X' },
    { _id: '2', title: 'AMD Ryzen 7900X' },
  ],
  manufacturers: [{ _id: '1', title: 'AMD' }],
};

export const SEARCH_SCHEME = [
  { type: 'products', labels: ['title'], url: '/products' },
  { type: 'manufacturers', labels: ['title'], url: '/manufacturers' },
];
