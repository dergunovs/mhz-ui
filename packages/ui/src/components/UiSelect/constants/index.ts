export const MODEL_VALUE = { _id: '1', title: 'Один', title_en: 'One' };

export const IS_FILTER = false;

export const OPTIONS = ['1', '2', '3', '4', '5', '6', '7', '8', '11', '111', '12'];

export const OPTIONS_OBJECTS = [
  { _id: '1', title: 'Один', title_en: 'One' },
  { _id: '2', title: 'Два', title_en: 'Two' },
  { _id: '3', title: 'Три', title_en: 'Three' },
];
