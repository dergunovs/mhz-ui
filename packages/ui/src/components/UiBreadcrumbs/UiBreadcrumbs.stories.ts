import { Meta, StoryObj } from '@storybook/vue3';
import { html } from 'mhz-helpers';

import { LINKS } from './constants';

import { UiBreadcrumbs } from '@/components';

const meta: Meta<typeof UiBreadcrumbs> = {
  component: UiBreadcrumbs,
  args: {
    links: LINKS,
  },
  argTypes: {
    color: {
      options: ['default', 'white'],
    },
  },
};

export default meta;

export const Primary: StoryObj<typeof UiBreadcrumbs> = {
  render: (args) => ({
    components: { UiBreadcrumbs },
    setup: () => ({ args }),

    template: html` <UiBreadcrumbs v-bind="args" />`,
  }),
};
