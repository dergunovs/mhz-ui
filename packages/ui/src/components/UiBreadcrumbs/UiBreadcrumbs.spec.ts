import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import { dataTest } from 'mhz-helpers';

import UiBreadcrumbs from './UiBreadcrumbs.vue';

import { DEFAULT_ARIA_LABEL, DEFAULT_COLOR, LINKS } from './constants';

import { wrapperFactory } from '@/test';

const breadcrumb = dataTest('ui-breadcrumb');
const breadcrumbLink = dataTest('ui-breadcrumb-link');
const breadcrumbTitle = dataTest('ui-breadcrumb-title');
const breadcrumbSlash = dataTest('ui-breadcrumb-slash');
const breadcrumbPosition = dataTest('ui-breadcrumb-position');

let wrapper: VueWrapper<InstanceType<typeof UiBreadcrumbs>>;

beforeEach(() => {
  wrapper = wrapperFactory(UiBreadcrumbs, { color: DEFAULT_COLOR, links: LINKS });
});

enableAutoUnmount(afterEach);

describe('UiBreadcrumbs', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(UiBreadcrumbs)).toBeTruthy();
  });

  it('matches snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('shows breadcrumbs', async () => {
    expect(wrapper.findAll(breadcrumb).length).toBe(LINKS.length);
  });

  it('shows breadcrumb title', async () => {
    expect(wrapper.findAll(breadcrumbTitle)[0].text()).toBe(LINKS[0].title);
  });

  it('sets breadcrumb position in meta', async () => {
    const index = 2;

    expect(wrapper.findAll(breadcrumbPosition)[index].attributes('content')).toBe((index + 1).toString());
  });

  it('generates links except last element', async () => {
    expect(wrapper.findAll(breadcrumbLink)[0].attributes('to')).toBe(LINKS[0].url);
    expect(wrapper.findAll(breadcrumbLink)[0].attributes('data-link')).toBe('true');

    expect(wrapper.findAll(breadcrumbLink)[LINKS.length - 1].attributes('data-link')).toBe('false');
  });

  it('shows slashes after links except last element', async () => {
    expect(wrapper.findAll(breadcrumbSlash).length).toBe(LINKS.length - 1);
  });

  it('changes color by props', async () => {
    expect(wrapper.findAll(breadcrumbLink)[0].attributes('data-color')).toBe(DEFAULT_COLOR);
    expect(wrapper.findAll(breadcrumbSlash)[0].attributes('data-color')).toBe(DEFAULT_COLOR);

    const newColor = 'white';

    await wrapper.setProps({ color: newColor });

    expect(wrapper.findAll(breadcrumbLink)[0].attributes('data-color')).toBe(newColor);
    expect(wrapper.findAll(breadcrumbSlash)[0].attributes('data-color')).toBe(newColor);
  });

  it('handles empty links array', async () => {
    await wrapper.setProps({ links: [] });

    expect(wrapper.findAll(breadcrumb).length).toBe(0);
    expect(wrapper.findAll(breadcrumbLink).length).toBe(0);
    expect(wrapper.findAll(breadcrumbSlash).length).toBe(0);
  });

  it('handles single link', async () => {
    await wrapper.setProps({ links: [LINKS[0]] });

    expect(wrapper.findAll(breadcrumb).length).toBe(1);
    expect(wrapper.findAll(breadcrumbLink).length).toBe(1);
    expect(wrapper.findAll(breadcrumbSlash).length).toBe(0);
  });

  it('sets aria-label for accessibility', async () => {
    expect(wrapper.attributes('aria-label')).toBe(DEFAULT_ARIA_LABEL);

    const newAriaLabel = 'Breadcrumb navigation';

    await wrapper.setProps({ ariaLabel: newAriaLabel });

    expect(wrapper.attributes('aria-label')).toBe(newAriaLabel);
  });
});
