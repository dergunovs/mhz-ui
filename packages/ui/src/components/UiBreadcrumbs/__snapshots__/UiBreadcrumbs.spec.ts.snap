// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UiBreadcrumbs > matches snapshot 1`] = `
"<div class="_breadcrumbs_9e0c71" itemscope="" itemtype="https://schema.org/BreadcrumbList">
  <div class="_links_9e0c71" itemprop="itemListElement" itemscope="" itemtype="https://schema.org/ListItem" data-test="ui-breadcrumb"><a to="/" class="_link_9e0c71" data-link="true" data-color="default" itemid="/" itemtype="https://schema.org/Thing" itemscope="" itemprop="item" data-test="ui-breadcrumb-link"><span itemprop="name" data-test="ui-breadcrumb-title">Main</span></a><span class="_slash_9e0c71" data-color="default" data-test="ui-breadcrumb-slash">/</span>
    <meta itemprop="position" content="1" data-test="ui-breadcrumb-position">
  </div>
  <div class="_links_9e0c71" itemprop="itemListElement" itemscope="" itemtype="https://schema.org/ListItem" data-test="ui-breadcrumb"><a to="/category" class="_link_9e0c71" data-link="true" data-color="default" itemid="/category" itemtype="https://schema.org/Thing" itemscope="" itemprop="item" data-test="ui-breadcrumb-link"><span itemprop="name" data-test="ui-breadcrumb-title">Categories</span></a><span class="_slash_9e0c71" data-color="default" data-test="ui-breadcrumb-slash">/</span>
    <meta itemprop="position" content="2" data-test="ui-breadcrumb-position">
  </div>
  <div class="_links_9e0c71" itemprop="itemListElement" itemscope="" itemtype="https://schema.org/ListItem" data-test="ui-breadcrumb">
    <div to="/cpu" class="_link_9e0c71" data-link="false" data-color="default" itemid="/cpu" itemtype="https://schema.org/Thing" itemscope="" itemprop="item" data-test="ui-breadcrumb-link"><span itemprop="name" data-test="ui-breadcrumb-title">CPU</span></div>
    <!--v-if-->
    <meta itemprop="position" content="3" data-test="ui-breadcrumb-position">
  </div>
</div>"
`;
